spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/ReadByFormatIntegrationTestBase.java

```diff
@@ -32,11 +33,17 @@
 
 public class ReadByFormatIntegrationTestBase extends SparkBigQueryIntegrationTestBase {
 
-  protected String dataFormat;
+  protected final String dataFormat;
+  protected final boolean userProvidedSchemaAllowed;
```

Comment:
Why do we need this?

Reply:
Yes - spark 3 does not support it properly for BigQuery